pkg/trie/triedb/triedb.go

```diff
@@ -29,14 +29,33 @@ type TrieDB struct {
 	rootHash common.Hash
 	db       db.DBGetter
 	cache    cache.TrieCache
+	layout   trie.TrieLayout
+	// rootHandle is an in-memory-trie-like representation of the node
+	// references and new inserted nodes in the trie
+	rootHandle NodeHandle
```

Comment:
nit/the handle naming is not intuitive to me, perhaps a different name would be more clear? Really like the comments though

Reply:
I added a comment in NodeHandle structure as a doc, let me know if that's self explanatory or if we still need to change the name. Also open to new name suggestions haha